enhancements/sig-architecture/20-taint-toleration/README.md

```diff
@@ -0,0 +1,346 @@
+
+### Examples
+
+#### 1. Cluster Maintaining
+
+Users (admins) can set a cluster to maintaining status, so workloads will be evicted and no workloads will be scheduled to the cluster. There's no need to set the toleration in placement.yaml under this senario. If toleration has to be set, make sure there won't be a matching key-value with the `maintaining` key-value in taint.
+
+cluster.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: cluster1
+  labels:
+    cluster.open-cluster-management.io/clusterset: clusterset1
+spec:
+  hubAcceptsClient: true
+  taints:
+    - key: maintaining
+      value: "true"
+```
+
+placement.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement1
+  namespace: default
+spec: ~
+```
+
+#### 2. Schedule on Certain Clusters
+
+Developers want to schedule some workloads to certain managed clusters.
+
+cluster.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: cluster1
+  labels:
+    cluster.open-cluster-management.io/clusterset: clusterset1
+spec:
+  hubAcceptsClient: true
+  taints:
+    - key: gpu
+      value: "true"
+```
+placement.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement1
+  namespace: default
+spec:
+  tolerations:
+    - key: gpu
+      value: "true"
+      operator: Equal
+```
+
+#### 3. Support Filtering Unhealthy/Unreachable Clusters
+
+When managed clusters' availability becomes `false`, the taint `open-cluster-management.io/unhealthy` would be automatically added to clusters. If managed clusters' availability becomes `unknown`, the taint `open-cluster-management.io/unreachabl` would be added. Those unhealthy or unreachable clusters could be evicted immediatly or after `TolerationSeconds`.
+
+##### 3.1 Immediate Eviction
+
+Admins or developers want applications deployed on this cluster to be transferred to another available managed cluster immediately.
+
+cluster.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: cluster1
+  labels:
+    cluster.open-cluster-management.io/clusterset: clusterset1
+spec:
+  hubAcceptsClient: true
+  taints:
+    - key: gpu
+      value: "true"
+    - key: unhealthy
+```
+
+placement.yaml:
+
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: Placement
+metadata:
+  name: placement1
+  namespace: default
+spec:
+  tolerations:
+    - key: gpu
+      value: "true"
+      operator: Equal
+```
+
+##### 3.2 Eviction After TolerationSeconds
+
+When a managed cluster gets offline, developers or admins want applications deployed on this cluster to be transferred to another available managed cluster after a tolerated time.
```

Comment:
In thinking about the mechanics of this, one possible way of handling this is as described in the API here where first the impacted placement is removed and then a new target chosen.  I could imagine someone wanting a way to surge and have a replacement before the removal.

This is an observation and not required for this iteration.